src/test/java/be/syntra/devshop/DevshopFront/controllers/PaginationControllerTest.java

```diff
@@ -0,0 +1,134 @@
+package be.syntra.devshop.DevshopFront.controllers;
+
+import be.syntra.devshop.DevshopFront.configuration.WebConfig;
+import be.syntra.devshop.DevshopFront.exceptions.JWTTokenExceptionHandler;
+import be.syntra.devshop.DevshopFront.models.SearchModel;
+import be.syntra.devshop.DevshopFront.models.dtos.CartDto;
+import be.syntra.devshop.DevshopFront.models.dtos.ProductsDisplayListDto;
+import be.syntra.devshop.DevshopFront.services.CartService;
+import be.syntra.devshop.DevshopFront.services.ProductService;
+import be.syntra.devshop.DevshopFront.services.SearchService;
+import be.syntra.devshop.DevshopFront.services.utils.ProductMapper;
+import be.syntra.devshop.DevshopFront.testutils.TestSecurityConfig;
+import be.syntra.devshop.DevshopFront.testutils.TestWebConfig;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+
+import static be.syntra.devshop.DevshopFront.testutils.ProductUtils.getDummyProductList;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest(controllers = PaginationController.class,
+        excludeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = {WebConfig.class, JWTTokenExceptionHandler.class})
+)
+@ContextConfiguration(classes = {TestWebConfig.class, TestSecurityConfig.class})
+@ExtendWith(MockitoExtension.class)
+public class PaginationControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @MockBean
+    private ProductMapper productMapper;
+
+    @MockBean
+    private ProductService productService;
+
+    @MockBean
+    private CartService cartService;
+
+    @MockBean
+    private SearchService searchService;
+
+    @InjectMocks
+    private PaginationController paginationController;
+
+    @ParameterizedTest
+    @ValueSource(strings = {"/pagination/previous", "/pagination/next", "/pagination/first", "/pagination/last/5"})
+    void canRequestFirstPreviousNextPages() throws Exception {
```

Comment:
if you want to test the ValueSource, you need to add as parameter: void canRequestFirstPreviousNextPages(String url) and change the to mockMvc.perform(get(url));

Reply:
fixed :-) passing tests :-)